File: Problem9/pathsum.java
class Solution {
    int max_sum;
    public int maxPath(TreeNode root)
    {
        if(root==null)
              return  0;
        int left_sum=maxPath(root.left);
        int right_sum=maxPath(root.right);
        int val=root.val+Math.max(left_sum,Math.max(right_sum,Math.max(0,left_sum+right_sum)));
        max_sum=Math.max(max_sum,val);
        return root.val+Math.max(0,Math.max(left_sum,right_sum));
    }
    public int maxPathSum(TreeNode root) {
        max_sum=Integer.MIN_VALUE;
        maxPath(root);
        return max_sum;
    }
}

File: Problem9/pathsum.py
from sys import last_value
from typing import Optional
class TreeNode:
    def __init__(self,val=-1,left=None,right=None):
        self.val=val
        self.left=left
        self.right=right
class Solution:
    max_val=0
    def maxPath(root)->int:
        if(root==None):
            return 0
        left_sum=Solution.maxPath(root.left)
        right_sum=Solution.maxPath(root.right)
        Solution.max_val=max(Solution.max_val,root.val+max(left_sum,right_sum,0,left_sum+right_sum))
        return root.val+max(left_sum,right_sum,0)

    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        Solution.max_val=pow(10,8)*-1
        Solution.maxPath(root)
        return Solution.max_val